learn/cookbooks/langchain.mdx

```diff
@@ -0,0 +1,187 @@
+
+<Capsule intent="tip">
+
+This guide uses Meilisearch’s Python SDK. But you can use Meilisearch with almost any language.
+
+</Capsule>
+
+## Requirements
+
+This tutorial assumes a basic understanding of Python and LangChain. Beginners to LangChain will still find the tutorial accessible.
+
+- Python (LangChain requires >= 3.8.1 and < 4.0) and the pip CLI
+- A Meilisearch >= 1.6 instance running — see our [quick start](/learn/getting_started/cloud_quick_start)
+- An OpenAI API key — [get yours](https://platform.openai.com/account/api-keys)
+
+## Setting up a Meilisearch instance
+
+After creating your Meilisearch instance, make sure to grab your Meilisearch host and an API key with write permissions.
+
+<Capsule intent="warning">
+
+In production, we recommend using different API keys holding only the minimal permissions to operate.
+
+</Capsule>
+
+
+Enable the vector store feature. On Meilisearch Cloud, enable it via your project’s Settings page. For self-hosted Meilisearch, activate the vector store with the [API route](/learn/experimental/vector_search#activate-vector-search).
+
+## Creating the project
+
+Create a folder for our project with an empty `main.py` file for your code.
+
+Before writing the code, install the necessary dependencies:
+
+```bash
+pip install langchain openai meilisearch python-dotenv
+```
+
+<Capsule intent="tip">
+
+This guide uses python-dotenv to load environment variables from a `.env` file. Feel free to load them in any other convenient way for you.
+
+</Capsule>
+
+If using dotenv, first create a .env to store our credentials:
+
+```
+# .env
+
+MEILI_HTTP_ADDR="your Meilisearch host"
+MEILI_API_KEY="your Meilisearch API key"
+OPENAI_API_KEY="your OpenAI API key"
+
+```
+Now that you have your environment variables available, create a `setup.py` file with some boilerplate code for your example:
+
+```python
+# setup.py
+
+import os
+from dotenv import load_dotenv # remove if not using dotenv
+from langchain.vectorstores import Meilisearch
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.document_loaders import JSONLoader
+
+load_dotenv() # remove if not using dotenv
+
+# exit if missing env vars
+if "MEILI_HTTP_ADDR" not in os.environ:
+    raise Exception("Missing MEILI_HTTP_ADDR env var")
+if "MEILI_API_KEY" not in os.environ:
+    raise Exception("Missing MEILI_API_KEY env var")
+if "OPENAI_API_KEY" not in os.environ:
+    raise Exception("Missing OPENAI_API_KEY env var")
+
+# Setup code will go here 👇
+```
+
+## Importing documents and embeddings
+
+Now that the project is ready, import some documents in Meilisearch. First, download this small movies dataset:
+
+[movies-lite.json](https://gist.github.com/Strift/1524ab5e2015e50bbcb215fb4d950a38)
```

Comment:
If you want users to download it, you can use an `<a>` tag with a couple of special attributes. Check out the self-hosted quick start for an example.

Also, I think it might look better if move the link so it's part of the previous paragraph.